scripts/flattener.py

```diff
@@ -656,24 +655,18 @@ def clean_var():
 
 
 # Reconcile genes if raw matrices annotated to multiple version by merging raw based on Ensembl ID
-def reconcile_genes(mfinal_obj, cxg_adata_lst, mfinal_adata_genes):
+# Return raw matrix merged on Ensembl ID and list of gene to drop from normalized matrices
+def reconcile_genes(cxg_adata_lst):
+	global mfinal_obj
+	global mfinal_adata
+	mfinal_adata_genes = mfinal_adata.var.index.to_list()
 	redundant = []
-	redundant_within_version = []
-	chain_redundant_genes = []
-	gene_multi_ensembl = []
-	versions_checked = []
-	genes_to_collapse_final = {}
-	collapsed_need_to_only_switch_id = {}
-	total_genes = []
+	multiple_ensembl = []
```

Comment:
I think that this could get cleaned up a little
`stats = {}`

then later, instead of the 3x `extend`, you could define the key,value
`stats['multiple_ensembl'] = multiple_ensembl`

then, you just need to adjust this line
`all_remove = list(set(redundant + multiple_ensembl + multiple_symbols))`
and call the dict
`all_remove = list(set(stats['redundant'] + stats['multiple_ensembl'] + stats['multiple_symbols']))`
or even use list comprehension
`all_remove = list(set([i for l in stats.values() for i in l]))`

Reply:
Good point. For some reason, I got it in my head that you needed to instantiate the value for each key if it is a list. I think I am getting confused with a nested dictionary.
And after looking at this comment, I realized I made a mistake. It doesn't change the outcome, but 'redundant' is actually a list of Ensembl IDs. So, it is dropped separately at a different step after the Ensembl IDs have been mapped. 

I'll make all the changes:
- drop 'redundant', 'multiple_ensembl', 'multiple_symbols' variables
- Just instantiate stats = {}
- all_remove should not include 'redundant'